workflows/TW-phoSim-r3/phoSimPrep.py

```diff
@@ -0,0 +1,79 @@
+#!/nfs/farm/g/lsst/u1/software/redhat6-x86_64-64bit-gcc44/anaconda/2.3.0/bin/python
+
+## phoSimPrep.py - Perform potentially lengthy preparatory steps prior to running phoSim
+##
+
+import os,sys,shutil
+
+## Insert task config area for python modules (insert as 2nd element in sys.path)
+sys.path.insert(1,os.getenv('TW_CONFIGDIR'))
+from config import *
+
+
+print '\n\nWelcome to phoSimPrep.py\n========================\n'
+
+## Generate instanceCatalog on-the-fly
+
+## Create scratch directory in (persistent) /lustre, if necessary.
+## File path = PHOSIMSCRATCH (defined in config.py)
+
+if not os.path.exists(PHOSIMSCRATCH): os.makedirs(PHOSIMSCRATCH)
+
+## ### Until actual code is available, simply copy/link necessary files
+## ### into pre-agreed scratch space.
+## base = 'phosim_input_200'
+
+## src = os.path.join(PHOSIMIN,base+'.txt.gz')
+## dst = os.path.join(PHOSIMSCRATCH,'instanceCatalog.txt.gz')
+## print 'Instance Catalog:'
+## print 'src = ',src
+## print 'dst = ',dst
+## shutil.copy2(src,dst)
+
+## src = os.path.join(PHOSIMIN,base+'.tar.gz')
+## dst = os.path.join(PHOSIMSCRATCH,'SEDs.tar.gz')
+## print 'SEDs:'
+## print 'src = ',src
+## print 'dst = ',dst
+## shutil.copy2(src,dst)
+
+## generate instance catalog and SED files for phoSim
+
+#  generatePhosimInput.py obsHistID [options]
+destIC = os.path.join(PHOSIMSCRATCH,'instanceCatalog.txt')
+destSEDdir = PHOSIMSCRATCH
+obsHistID = os.getenv('TW_OBSHISTID')
+print 'destIC = ',destIC
+print 'destSEDdir = ',destSEDdir
+print 'obsHistID = ',obsHistID
+
+##       SCRIPT to generate phoSim instance catalog and custom SED library
+##
+# generatePhosimInput.py <opsHistID> --OpSimDBDir <loc-opsSim database> --seddir <loc-to-write-output SEDs> --outfile <instanceCat-name>
+### e.g., (the SEDs will occupy the "spectra_files" directory)
+# generatePhosimInput.py 200 --OpSimDBDir /nfs/farm/g/desc/u1/data/Twinkles --seddir . --outfile phosim_input_200.txt
+
+genCmd = 'generatePhosimInput.py'
+
+print '\nGenerate InstanceCatalog and SED files'
+cmd = "time "+genCmd+" "+obsHistID+" --OpSimDBDir /nfs/farm/g/desc/u1/data/Twinkles --seddir "+destSEDdir+" --outfile "+destIC
```

Comment:
With the merges into master of #380 and #381, the calling sequence for `generatePhosimInput.py` has changed.  To take advantage of the local caches of the star, galaxy, and SN dbs, the `--cache_dir` option should be added:
```
 generatePhosimInput.py 230 --OpSimDBDir /nfs/farm/g/desc/u1/data/Twinkles --seddir <seddir> --outfile <outfile> --cache_dir /nfs/farm/g/desc/u1/data/Twinkles/fatboy_caches 
```
In this example, I've put the current locations of the `minion_1016_sqlite.db` file (`/nfs/farm/g/desc/u1/data/Twinkles`) and the `*_cache.db` files (`/nfs/farm/g/desc/u1/data/Twinkles/fatboy_caches`).  If those files are copied to local scratch space to avoid collisions with other processes trying to access them at the same time, those paths should of course be updated.